color_windows.go

```diff
@@ -0,0 +1,17 @@
+package color
+
+import (
+	"os"
+
+	"golang.org/x/sys/windows"
+)
+
+func init() {
+	// Opt-in for ansi color support for current process.
```

Comment:
How to opt-out in case it causes issues?

Reply:
The "opt-in" comment is in relation to the Microsoft documentation, which calls it to "opt in for color support".
See how ENABLE_VIRTUAL_TERMINAL_PROCESSING is used in their examples at https://learn.microsoft.com/en-us/windows/console/console-virtual-terminal-sequences#output-sequences for a better understanding.

I can not comment on possible issues this would cause, as I had no issues with it. I would just argue that it is by recommendation from Microsoft.

Reply:
Additionally, see this PR for crossterm, which is a rust crate with color support that dealt with the same issue last year:

https://github.com/crossterm-rs/crossterm/pull/657

Reply:
We have a similiar constraint in the package that automatically disabled the coloring based on the environment, see: https://github.com/fatih/color/blob/main/color.go#L22

I wonder if we could move this into a function, and move it there. I think we still want to keep `color_windows.go` though.

Reply:
> How to opt-out in case it causes issues?

I think that in "no color" mode this Windows API should not be called. 

The reason is that this code changes the console mode. So the console would have a different "behavior" after the program using `github.com/fatih/color` exits. 

Moreover, there should be a way to revert to the original value of ConsoleMode.

Personally, I propose to simply add this as a code snippet in `README.md` as a workaround for https://github.com/fatih/color/issues/169

1. It is "safer". I think it is better to not break anyone by changing the current console mode just to enable coloring.
2. It does not seem like a frequent issue.
3. It does not bring additional dependency 😉

Reply:
I think we should include this into the library. I like adding instructions to Readme, but this is a fundamental fix, and not including it breaks coloring for people using Windows. So we should make it a priority for them.

Reply:
@pellared and @fatih 

I re-did this PR as #186 because I accidentally deleted the source fork. I still didn't address your last 2 comments. Is there a consensus about opting out on this?

Personally, I think it is uneeded because the majority of Windows users will not exercise this code anyway as they will not be running in a 3rd party shell. Windows Terminal and Windows Powershell both does this "opt in" on color themselves, so apps there don't need to perform the signal. Also it does not seem neither of those offer an easy "opt out" on coloring.

Reply:
Yeah, so I think this should be opt-in. It's only for people who use windows and if it causes an issue, we can always push a bugfix version. I like option for customization, but not to make things work the way it should. So if coloring doesn't work, nobody should be hunting down configuration in the readme to make it work. It's not very user friendly.